packages/web/src/components/user-name-and-badges/UserNameAndBadges.tsx

```diff
@@ -45,3 +56,23 @@ export const UserNameAndBadges = (props: UserNameAndBadgesProps) => {
     </ArtistPopover>
   )
 }
+
+const LoadUserAndRender = (props: UserNameAndBadgesWithIdProps) => {
+  const currentUserId: number = useSelector(getUserId)!
+  const { data: user } = useGetUserById({ id: props.userId, currentUserId })
+  return <UserNameAndBadgesImpl {...props} user={user} />
+}
+
+function isIdProps(
+  props: UserNameAndBadgesProps
+): props is UserNameAndBadgesWithIdProps {
+  return (props as UserNameAndBadgesWithIdProps).userId != null
+}
+
+export const UserNameAndBadges = (props: UserNameAndBadgesProps) => {
```

Comment:
this pattern feels code smelly. it feels like it should just always be ID driven and the selector is ~free if you already have data somewhere, but no need to change in this PR

Reply:
I would agree! But we use this component in a lot of other places and I wanted to leave it backwards compatible.